packages/core-ethereum/src/channel.ts

```diff
@@ -98,17 +98,28 @@ class Channel {
     return (await this.themToUs()).commitment
   }
 
+  getUsToThemId(): Hash {
+    return generateChannelId(this.self.toAddress(), this.counterparty.toAddress())
+  }
+
   async usToThem(): Promise<ChannelEntry> {
-    return await this.indexer.getChannel(generateChannelId(this.self.toAddress(), this.counterparty.toAddress()))
+    try {
+      return await this.indexer.getChannel(this.getUsToThemId())
+    } catch {
```

Comment:
I prefer this over having the generic error, fine with me eitherway

Reply:
the problem with this is that it doesn't check that the error is actually db missing the entry. If we had a syntax error in db for example, this would swallow it and we'd get the wrong error message. catch and rethrow is usually a bad idea, unless we are specifically testing the type of error.

Reply:
thanks! good point, fixing